### PR TITLE
Remove labels without classes info from the export modal

### LINKED ISSUE #822: Dataset export is failing with large datasets
## Describe the bug

When a dataset is too big, the `countLabelsOfDataset` query timeouts as it can send thousands of queries.

* [One of the failing dataset](https://labelflow.ai/ores/datasets/ores_namur-zone-ciney-biron-ckhonn24908dw0702toud6fxc/images)
* [Additional details](https://labelflow.slack.com/archives/C022PCC0H0W/p1643014826075900)

## To Reproduce

1. Have a very large dataset
2. Export
3. 💣

## Expected behavior

Export should work without failures

## Screenshots

image
image

## Device information

 
N/A

## Additional context

With online workspaces

### Short term
- [ ] Don't display number of labels if not really needed.
- [ ] If data is actually needed on client-side, implement a dedicated GraphQL query which batches the SQL queries instead of relying on the apollo dynamic field resolution system (which introduces the n+1 problem).
- [ ] See if a similar query is not already used when the export is actually being made.

### Long term
- [ ] Use [dataloader](https://www.npmjs.com/package/dataloader)

### PR DESCRIPTION
**Pull request by Jordanlelay**
## Work performed

- Remove label without classes info from the export modal 
- Set black version in CI

## Resolved issues

closes #822

### CODE REVIEW

File: typescript/web/src/components/export-button/export-modal/modal-header.tsx [hunk 1]
```diff
@@ -24,27 +23,14 @@ export const ModalHeader = () => {
...
-            )
-        }
+      <Box width="100%" pl={8} pr={8} pt={4}>
+        <Alert status="info" borderRadius={5}>
+          <AlertIcon />
+          <AlertTitle fontSize={15}>About your export</AlertTitle>
+          <AlertDescription fontSize={13} fontWeight="medium">
+            Labels without a class are not exported
+          </AlertDescription>
+        </Alert>
```
Comment (bbenoist) [inline typescript/web/src/components/export-button/export-modal/modal-header.tsx:33]: I'll be displayed for every user. Even when they don't have any label with a missing label-class.
Do we really need to show this message with such an emphasis?

File: typescript/web/src/components/export-button/export-modal/modal-header.tsx [hunk 2]
```diff
@@ -24,27 +23,14 @@ export const ModalHeader = () => {
...
-                  } ${
-                    numberUndefinedLabelsOfDataset === 1 ? "has" : "have"
-                  } no class assigned. Only labels with a class
-            are exported.`}
-                </AlertDescription>
-              </Alert>
-            )
-        }
+      <Box width="100%" pl={8} pr={8} pt={4}>
+        <Alert status="info" borderRadius={5}>
```
Comment (bbenoist) [inline typescript/web/src/components/export-button/export-modal/modal-header.tsx:27]: While refactoring it, I'd have taken the opportunity to move this block in it's own `Disclaimer` component (_or `LabelWithoutLabelClassDisclaimer` if needed_).

**Review by bbenoist — APPROVED** [review]
Code looks OK.
I've made remarks but they're not blocking the merge.

File: typescript/web/src/components/export-button/export-modal/modal-header.tsx (hunk 1 above)
Comment (Jordanlelay) [reply typescript/web/src/components/export-button/export-modal/modal-header.tsx:33]: Maybe, maybe not. We can also completely remove it.

File: typescript/web/src/components/export-button/export-modal/modal-header.tsx (hunk 2 above)
Comment (Jordanlelay) [reply typescript/web/src/components/export-button/export-modal/modal-header.tsx]: Fixed

File: typescript/web/src/components/export-button/export-modal/modal-header.tsx (hunk 1 above)
Comment (nicOwlas) [reply typescript/web/src/components/export-button/export-modal/modal-header.tsx]: Let's remove it completely 👍

Comment (Jordanlelay) [reply typescript/web/src/components/export-button/export-modal/modal-header.tsx]: I have removed it

**Review by bbenoist — APPROVED** [review]

**Comment by bbenoist** [comment]
@Jordanlelay There is a weird issue with Python linting blocking this PR from being merged.
I doubt that it's because of the changes you've made but still, we're blocked.

Have you already looked at it or should we create an issue and start working on it ASAP?

**Comment by Jordanlelay** [comment]
> @Jordanlelay There is a weird issue with Python linting blocking this PR from being merged. I doubt that it's because of the changes you've made but still, we're blocked.
> 
> Have you already looked at it or should we create an issue and start working on it ASAP?

I guess it's a mismatch between locally installed version of black and the CI one. I will force the same version in the CI as in the requirements

**Comment by bbenoist** [comment]
You've merged the PR with new changes which I didn't reviewed 😢

**Comment by Jordanlelay** [comment]
> You've merged the PR with new changes which I didn't reviewed 😢

All required changes had passed and you approved it
If something is problematic, we can still revert it

**Comment by bbenoist** [comment]
I've approved the changes at a specific revisions but I've not reviewed the fix for the Python linter since it should have been made in a separate PR.
As we have bigger priorities, we can keep the changes as-them for now. But the version pinning should have been declared in a file shared between developers and CI (i.e. like what we have for `package.json`+`yarn.lock`), probably in the IOG `requirements.txt` file or similar.

**Pull request merged**